skel/lib/systemd/system-generators/dcache-generator

```diff
@@ -35,4 +42,6 @@ for domain in $(getProperty dcache.domains); do
 	[Install]
 	WantedBy=default.target
 	EOF
+
+    ln -s "$1/dcache@$domain.service" "$wantdir/dcache@$domain.service"
```

Comment:
I wouldn't mind if you renamed the generated unit to use a dash rather than an at symbol. As you pointed out, the purpose of the at symbol is different from what we do here.

Reply:
While this would make life possibly easier for us for now... I don't think this is the right way to go.

We should in fact have instance units here... as all the domains are instances of dCache (just doing different things).
We should IMO rather go the hard way, pull out the settings from dCache which are anyway not ought to be there (in the systemd world) and then do it with instance units and "@".

Reply:
Sure sure - but that's not going to fix the issue for 3.2 right now. As I said, I don't care much what you do in 3.3. We need a fix for 3.2 now. I realise it is my fault that what we got doesn't work. I would appreciate if you could help with this simple fix for 3.2 now - what you do in the future is up to you.

I am not trying to argue anything you are saying about what we should do in the next release. I just want a solution for the issue that right now prevents update to 3.2 on Debian.

Reply:
The fix for 3.2 is simply to get the global dcache.service file... this gives people back the functionality they had before: have everything started or nothing.
It just lacks what's advertised for in the release notes (that one can selectively enable/disable domains).

I don't think it's your fault... I think your solution is actually pretty elegant, and I have no idea why systemd puts obstacles in the way not supporting this :-(

And I don't argue either :-) I'm just trying to find out which is the way we go for 3.3.
I think for 3.2 the best thing is to simply merge the global unit file... and remove the advertisement for per-domain enable/disable in the release notes.

Reply:
You mean merge the global unit file without updating the generator to create the symlinks? How would that work?

Reply:
No, sorry, I wasn't clear:
I meant "for 3.2: merge this very pull request"... which gives the global unit file, and the necessary changes to the generator